### ISSUE TITLE
Refactor EonPublicKey database schema in Collator

### ISSUE THREAD

**Issue by ezdac** [shutter-network/rolling-shutter#256]
The database for memorising eon-pukeys from the keypers might be worth refactoring.
This makes most sense after #238 is solved completely, since the database schemas might have to be changed anyways,
and this takes away some ambiguity for duplicates, simplifying the query logic.

Question:
- Why are eon-public keys blindly inserted in the collators "eon" database?
- Why do we validate the threshold reached criterium on retrieving eon-public-key-messages and not on inserting them.

Some input on the issue:

- *"Those are only candidate eon-keys. The table is just used there to store the additional information from on-chain (threshold specified by keyper-config) in combination with the "proposed" eon-key.
The key is only considered to be valid/confirmed when it is returned from the GetEonPublicKeyMessages query:
This query does the num_messages >= threshold checking with the information in the Candidate-Eon table.
So to summarise: the handler stupidly inserts the messages in the DB while the validity-check is done at the time of querying the db for EonPublicKey-messages."*

- *"I think i will rename the eon table to candidate_eon because this is confusing"*
- *"How about leaving the key blank in that table until it's confirmed?"*

**Comment by schmir-at-bb** [comment]
3869a8149f9ece59132ef3cf09b1bcabda398fae concludes the work on this. most changes merged in  28e47a9d60ece6d174abc60055e4597ecbee873e
